kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java

```diff
@@ -162,6 +171,88 @@ void upstreamUsesTlsInsecure(@Tls KafkaCluster cluster) {
+        var clientCert = new KeytoolCertificateGenerator();
+        clientCert.generateSelfSignedCertificateEntry("clientTest@kroxylicious.io", "client", "Dev", "Kroxylicious.ip", null, null, "US");
+
+        try (var cluster = KafkaClusterFactory.create(KafkaClusterConfig.builder()
+                .brokerKeytoolCertificateGenerator(brokerCert)
+                // note passing client generator causes ssl.client.auth to be set 'required'
+                .clientKeytoolCertificateGenerator(clientCert)
+                .kraftMode(true)
+                .securityProtocol("SSL")
+                .build())) {
+            cluster.start();
+
+            var config = new HashMap<>(cluster.getKafkaClientConfiguration());
+
+            // Get the client key material provided by the framework. This will be used to configure Kroxylicious.
+            var keyStore = config.get("ssl.keystore.location").toString();
+            var keyPassword = config.get("ssl.keystore.password").toString();
+            assertThat(keyStore).isNotNull();
+            assertThat(keyPassword).isNotNull();
+
+            var trustStore = config.get("ssl.truststore.location").toString();
+            var trustPassword = config.get("ssl.truststore.password").toString();
+            assertThat(trustStore).isNotNull();
+            assertThat(trustPassword).isNotNull();
+
+            // Validate a TLS client-auth connection to direct the cluster succeeds/fails as expected.
+            assertSuccessfulDirectClientAuthConnectionWithClientCert(cluster);
+            assertUnsuccessfulClientAuthDirectConnectionWithoutClientCert(cluster);
+
+            var builder = new ConfigurationBuilder()
+                    .addToVirtualClusters("demo", new VirtualClusterBuilder()
+                            .withNewTargetCluster()
+                            .withBootstrapServers(cluster.getBootstrapServers())
+                            .withNewTls()
+                            .withNewTrustStoreTrust()
+                            .withStoreFile(trustStore)
+                            .withNewInlinePasswordStoreProvider(trustPassword)
+                            .endTrustStoreTrust()
+                            .withNewKeyStoreKey()
+                            .withStoreFile(keyStore)
+                            .withNewInlinePasswordStoreProvider(keyPassword)
+                            .endKeyStoreKey()
+                            .endTls()
+                            .endTargetCluster()
+                            .withClusterNetworkAddressConfigProvider(CONFIG_PROVIDER_DEFINITION)
+                            .build());
+
+            try (var tester = kroxyliciousTester(builder); var admin = tester.admin("demo")) {
+                // do some work to ensure connection is opened
+                final var result = admin.describeCluster().clusterId();
+                assertThat(result).succeedsWithin(Duration.ofSeconds(10));
```

Comment:
It'd have been stronger to do something to assert that the client's principal was actually been picked up by Kafka.  However, I don't see an easy way to do this.  ACL rules would work but that brings quite a lot of additional complexity to the test.  I think this  gets over the good enough hurdle.